src/components/UI/market/ProductForm.tsx

```diff
@@ -1,26 +1,30 @@
 import styled from 'styled-components';
 import axios from 'axios';
-import React, { useState } from 'react';
-import { useNavigate } from 'react-router-dom';
-import ProductInput from './ProductInput';
-import { postProduct } from '../../../api/market';
+import React, { useState, useEffect } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { postProduct, updateProduct } from '../../../api/market';
 import useToken from '../../../hooks/useToken';
+import ProductInput from './ProductInput';
+import { ProductDetail, ProductFormValues } from '../../../types';
+import { categories } from '../../../assets/data/categories';
 
-export interface ProductFormValues {
-  title: string;
-  price: number | string;
-  content: string;
+interface ProductFormProps {
+  productObj?: ProductDetail | undefined;
 }
 
-const ProductForm = () => {
+const ProductForm: React.FC<ProductFormProps> = ({ productObj }) => {
   const [categoryId, setCategoryId] = useState(0);
-  const [formData, setFormData] = useState<ProductFormValues>({
-    title: '',
-    price: '',
-    content: '',
-  });
+
   const token = useToken();
   const navigate = useNavigate();
+  const location = useLocation();
+
+  const [formData, setFormData] = useState<ProductFormValues>({
+    title: productObj ? productObj.title : '',
+    price: productObj ? productObj.price : '',
+    content: productObj ? productObj.content : '',
+    categoryId: productObj ? productObj.categoryId : categoryId,
+  });
```

Comment:
코드를 simplify 해볼 수도 있을듯?
근데 아래의 코드가 반드시 좋은 것만은 아니라서, 이럴 수도 있다 정도로 봐두면 될듯해

```suggestion
  const { title = '', price = '', content = '', categoryId = '' } = productObj || {};

  const [formData, setFormData] = useState<ProductFormValues>({
    title,
    price,
    content,
    categoryId,
  });
```

Reply:
그리고 아래에서 useEffect 를 통해서 productObj 가 바뀔 때마다 set 을 할거면 굳이 이렇게 힘들게 initialValue 를 넣을 필요가 없는데..?

Reply:
헉............!!!!@@@!!!!!!!!!!! 수정 완완 했습니다